fgpyo/util/metric.py

```diff
@@ -175,7 +175,17 @@ def parse(cls, fields: List[str]) -> Any:
         parsers = cls._parsers()
         header = cls.header()
         assert len(fields) == len(header)
-        return inspect.attr_from(cls=cls, kwargs=dict(zip(header, fields)), parsers=parsers)
+        return inspect.attr_from(
+            cls=cls, kwargs=dict(zip(header, map(cls.parser_formatting, fields))), parsers=parsers
```

Comment:
prefer a list comprehension since it is a lot clearer what it's doing, and there's almost never a performance degredation.
```suggestion
            cls=cls, kwargs=dict(zip(header, [cls.parser_formatting(field) for field in fields])), parsers=parsers
```